packages/api/src/command/medical/patient/settings/create-patient-settings.ts

```diff
@@ -138,50 +161,52 @@ export async function upsertPatientSettingsForCx({
 }: PatientSettingsUpsertForCxProps): Promise<PatientSettingsUpsertResults> {
   const { log } = out(`upsertPatientSettingsForCx - cx ${cxId}`);
 
-  const totalPatients = await getPatientsCount({ cxId, facilityId });
-  if (totalPatients === 0) {
+  const patientIds = await getPatientIds({ cxId, facilityId });
+  if (patientIds.length === 0) {
     log(`No patients found for cx ${cxId}`);
     return { patientsFoundAndUpdated: 0 };
   }
 
   let processedTotal = 0;
   let failedTotal = 0;
-  let fromItem: string | undefined;
-
-  do {
-    const { currPageItems, nextPageItemId } = await getPaginationItems(
-      { count: BATCH_SIZE, ...(fromItem ? { fromItem } : {}) },
-      pagination => getPatients({ cxId, facilityId, pagination }),
-      () => Promise.resolve(totalPatients)
-    );
-
-    if (currPageItems.length === 0) break;
+  const failedIds: string[] = [];
 
-    const patientIds = currPageItems.map(p => p.id);
-    const { processedCount, failedCount, error } = await processBatch({
-      patients: patientIds,
+  for (let i = 0; i < patientIds.length; i += BATCH_SIZE) {
+    const batchIds = patientIds.slice(i, i + BATCH_SIZE);
+    const {
+      processedCount,
+      failedCount,
+      failedIds: failedBatchIds,
+    } = await processBatch({
+      patientIds: batchIds,
       cxId,
       settings,
+      log,
     });
 
     processedTotal += processedCount;
     failedTotal += failedCount;
-    fromItem = nextPageItemId;
-
-    log(
-      `Processed batch: ${processedCount} successful, ` +
-        `${failedCount} failed. Total progress: ${processedTotal}/${totalPatients}`
-    );
-
-    if (error) {
-      log(`Error processing batch: ${error.message}`);
-    }
-  } while (fromItem);
+    failedIds.push(...(failedBatchIds ?? []));
+  }
 
   log(
     `Completed processing all patients. ` +
       `Total: ${processedTotal} successful, ${failedTotal} failed`
   );
+
+  if (failedTotal > 0) {
+    const msg = `Failed to upsert settings for patients`;
+    log(`${msg} - failed IDs: ${failedIds}`);
+    capture.error(msg, {
+      extra: {
+        failedIds,
+        cxId,
+        facilityId,
```

Comment:
@coderabbitai the order or properties inside `capture.error` and `capture.message`'s `extra` parameter matters: this is passed on to Sentry, and if we send too much data, Sentry will trim out the data beyond it's payload limit. We need to be careful to start with the most important information, in this case the `cxId`. The `facilityId` really can be the last one (not in all cases, though).

@